src/info/mod.rs

```diff
@@ -364,4 +373,135 @@ mod tests {
+                    vec![String::from("ref1"), String::from("ref2")],
+                ),
+            },
+            pending: PendingInfo {
+                pending_changes: String::from("info test pending pending_changes"),
+            },
+            version: VersionInfo {
+                version: String::from("info test version version"),
+            },
+            created: CreatedInfo {
+                creation_date: String::from("info test created creation_date"),
+            },
+            languages: LanguagesInfo {
+                languages_with_percentage: vec![LanguageWithPercentage {
+                    language: Language::Go,
+                    percentage: 100_f64,
+                }],
+                true_color: false,
+                info_color: DynColors::Ansi(AnsiColors::White),
+            },
+            dependencies: DependenciesInfo {
+                dependencies: String::from("info test dependencies dependencies"),
+            },
+            authors: AuthorsInfo {
+                authors: vec![Author::new(
+                    git_repository::bstr::BString::from("author1 name"),
+                    git_repository::bstr::BString::from("author1 email"),
+                    100,
+                    100,
+                    false,
+                )],
+                info_color: DynColors::Ansi(AnsiColors::White),
+            },
+            last_change: LastChangeInfo {
+                last_change: String::from("info test last_change last_change"),
+            },
+            contributors: ContributorsInfo {
+                number_of_contributors: 100,
+                number_of_authors_to_display: 100,
+            },
+            repo: UrlInfo {
+                repo_url: String::from("info test repo repo_url"),
+            },
+            commits: CommitsInfo {
+                number_of_commits: String::from("info test commits number_of_commits"),
+            },
+            lines_of_code: LocInfo { lines_of_code: 100 },
+            size: SizeInfo {
+                repo_size: String::from("info test size repo_size"),
+                file_count: 100,
+            },
+            license: LicenseInfo {
+                license: String::from("info test license license"),
+            },
+            disabled_fields: vec![],
+            text_colors: TextColors {
+                title: DynColors::Ansi(AnsiColors::White),
+                tilde: DynColors::Ansi(AnsiColors::White),
+                underline: DynColors::Ansi(AnsiColors::White),
+                subtitle: DynColors::Ansi(AnsiColors::White),
+                colon: DynColors::Ansi(AnsiColors::White),
+                info: DynColors::Ansi(AnsiColors::White),
+            },
+            no_color_palette: true,
+            no_bold: true,
+            dominant_language: Language::Go,
+            ascii_colors: vec![DynColors::Ansi(AnsiColors::White)],
+        };
```

Comment:
Why didn't you make use of `git-testtools` here 🤔 ? This seems like a good use case

Reply:
The one argument I can see is that it's more of a unit test to simply have the test be "with the given `Info`, does it display correctly?" Parsing an actual git repository is closer to an integration test ("Do we read a git repository *and* display its parsed information correctly?").

Reply:
Right, I definitely had in mind an integration test for the `Info` struct - which is more or less just a container of `InfoFields` - which are individually unit tested. All that setup just to unit test the `fmt` method feels like a missed opportunity IMO.

Reply:
Thank you for the comments, wasn't sure how tight the formatting needed to be. This is an extreme of the code as implemented is absolutely correct and any changes are flagged as failures.

Will revisit this testing and explore using `git-testtools`.
[62% covered ](https://app.codecov.io/gh/o2sh/onefetch/blob/add-test-info-mod-rs/src/info/mod.rs)

Reply:
If we ever want to *unit* test `Info` a bit more, I think it would be a good idea to implement `Default` on it. Then we can just unit test like so:
```rust
#[test]
fn test_foo() {
    let info = Info { foo: value, ..Default::default() };
    // ...
}
```

Reply:
lol, fun to code, but not what is needed, resolving in favor of more readable code and info structure created from real repo.